src/java/htsjdk/samtools/CRAMIterator.java

```diff
@@ -83,10 +83,10 @@ public void setValidationStringency(
     public CRAMIterator(final InputStream inputStream, final ReferenceSource referenceSource)
             throws IOException {
         if (null == referenceSource) {
-            throw new CRAMException("A reference source is required for CRAM files");
-        }
+            this.referenceSource = new ReferenceSource();
+        } else
+            this.referenceSource = referenceSource;
```

Comment:
In this particular case (reading via CRAMIterator), you'll get a CRAMException "Contig  nnn not found in reference file" if no reference is provided :
	at htsjdk.samtools.CRAMIterator.hasNext(CRAMIterator.java:257)
	at htsjdk.samtools.SamReader$AssertingIterator.hasNext(SamReader.java:567)
	at org.broadinstitute.hellbender.utils.iterators.SAMRecordToReadIterator.hasNext(SAMRecordToReadIterator.java:24)

The NPE happens when you *write* a CRAM file with no reference:

java.lang.RuntimeException: java.lang.NullPointerException
	at htsjdk.samtools.CRAMFileWriter.writeAlignment(CRAMFileWriter.java:339)
	at htsjdk.samtools.SAMFileWriterImpl.addAlignment(SAMFileWriterImpl.java:180)

IIRC this second stack represents an exception that has already been caught and re-thrown so it doesn't reflect where the NPE originated (which I think was due to a call to getReferenceBases that returned null). As a side note I'd like to try to narrow the exception handling and propagation in the CRAM code, especially in this file around the hasNext/nextContainer code.